addon/components/bs-navbar.js

```diff
@@ -230,28 +231,18 @@ export default class Navbar extends Component {
    * @event onExpand
    * @public
    */
-  onExpand() {}
+  get onExpand() {
+    return this.args.onExpand ?? (() => {});
+  }
 
   /**
    * The action to be sent after the navbar has been expanded (including the CSS transition).
    *
    * @event onExpanded
    * @public
    */
-  onExpanded() {}
-
-  @observes('_collapsed')
-  _onCollapsedChange() {
-    let collapsed = this._collapsed;
-    let active = this.active;
-    if (collapsed !== active) {
-      return;
-    }
-    if (collapsed === false) {
-      this.show();
-    } else {
-      this.hide();
-    }
```

Comment:
I struggle to understand the impact of this change. Especially I struggle to understand what this code was doing _before_ the change.

* Are `show` and `hide` methods defined somewhere?
* `active` seems to be always `undefined`.

Maybe it was deadcode, which was always stepping out at line 248 as `_collapsed` was a boolean and `active` was always `undefined`? It seems that `collpased !== active` always returned `true` causing the observe to do nothing in practice.

Does that match your findings? Or do I miss something?

Reply:
This was a pretty tough one. I could not find any documentation on the existence of `this.show` and `this.hide`. Neither in the Ember docs or in the code itself. It seems like these methods just did not exist at all. Similar methods do exist in `bs-collapse` however. Using `console.log` on these functions (and `this.active`) seems to confirm that they are undefined. It is possible in this implementation that someone would add these 3 things through the component parameters though...

For the current implementation, everything relies on derrived data using
```js
  get collapsed() {
    return this._toggledCollapse ?? this.args.collapsed ?? true;
  }
  ```
  With priority being given to manual toggle.

I based the rules of the implementation mostly on the tests